arches/app/models/migrations/8009_etlmodule.py

```diff
@@ -128,17 +128,35 @@
     CREATE OR REPLACE FUNCTION public.__arches_load_staging_report_errors(load_id uuid)
     RETURNS TABLE(source text, message text, loadid uuid)
     AS $$
-    SELECT source_description, public.__arches_load_staging_get_tile_errors(value) AS message, loadid
-    FROM load_staging
-    WHERE passes_validation IS NOT true
-    AND loadid = load_id;
-    $$
-    LANGUAGE SQL;
+        SELECT source_description, CONCAT_WS (' | ', public.__arches_load_staging_get_tile_errors(value), error_message) AS message, loadid
+        FROM load_staging
+        WHERE passes_validation IS NOT true
+        AND loadid = load_id;
+    $$ LANGUAGE SQL;
+
+    CREATE OR REPLACE PROCEDURE public.__arches_check_tile_cardinality_violation_for_load(load_id uuid)
+    AS $$
+        UPDATE load_staging
+            SET error_message = 'excess tile error', passes_validation = false
+            WHERE (resourceid, nodegroupid, COALESCE(parenttileid::text, '')) IN (
+                SELECT t.resourceinstanceid, t.nodegroupid, COALESCE(t.parenttileid::text, '')
+                    FROM tiles t, node_groups ng1
+                    WHERE t.nodegroupid = ng1.nodegroupid
+                    AND ng1.cardinality = '1'
+                UNION
```

Comment:
The SELECT statement on line 147 doesn't have the join to the nodegroups table, so they aren't filtered by cardinality. I think the following update would ensure cardinality n nodegroups aren't included:
```
	SELECT resourceid, ng1.nodegroupid, COALESCE(parenttileid::text, '')
		FROM load_staging ls, node_groups ng1
		WHERE ls.nodegroupid = ng1.nodegroupid
		AND ng1.cardinality = '1'
		GROUP BY resourceid, ng1.nodegroupid, COALESCE(parenttileid::text, ''), loadid
		HAVING count(*) > 1
		AND loadid = ls.loadid
```